packages/react-router/lib/hooks.tsx

```diff
@@ -155,16 +185,8 @@ export function useNavigate(): NavigateFunction {
   let { matches } = React.useContext(RouteContext);
   let { pathname: locationPathname } = useLocation();
 
-  // Ignore index + pathless matches
-  let pathContributingMatches = matches.filter(
-    (match, index) =>
-      index === 0 ||
-      (!match.route.index &&
-        match.pathnameBase !== matches[index - 1].pathnameBase)
-  );
-
   let routePathnamesJson = JSON.stringify(
-    pathContributingMatches.map((match) => match.pathnameBase)
+    getPathContributingMatches(matches).map((match) => match.pathnameBase)
```

Comment:
I had a feeling doing the `JSON.stringify` inline in both spots was where it would end up.

Reply:
yeah it certainly worked fine, just felt a little distanced in the shared util since it's a dependency implementation detail for the individual hooks